### PR TITLE
CIRCSTORE-398 Add effective call number, shelving order and  pickup service point name fields to the request schema

### PR DESCRIPTION
**Pull request by tatsiana-tarhonskaya**
Resolves [CIRCSTORE-398](https://issues.folio.org/browse/CIRCSTORE-398)

## Purpose
Add effective call number, shelving order and service point name fields to the request schema

## Approach
Update schema and add corresponding indexes:

- gin indexes for callNumber and fullCallNumber (for now they're going to be searched as 'starts with' but there're plans to add 'contains' search in future)
- b-tree indexes for shelving order and pickup service point name  - they're going to be used for sorting by these fields

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <modelVersion>4.0.0</modelVersion>
   <artifactId>mod-circulation-storage</artifactId>
   <groupId>org.folio</groupId>
-  <version>16.1.0-SNAPSHOT</version>
+  <version>16.1.1-SNAPSHOT</version>
```
Comment (alexanderkurash) [inline pom.xml:5]: No need to change module version here, but interface version needs to be changed

**Review by alexanderkurash — CHANGES_REQUESTED** [review]

File: ramls/request.json [hunk 2]
```diff
@@ -226,6 +226,11 @@
       "type": "string",
       "format": "date-time",
       "readonly" : true
+    },
+    "searchIndex": {
+      "description": "Request fields used for search",
+      "type": "object",
+      "$ref": "searchIndex.json"
```
Comment (alexanderkurash) [inline ramls/request.json:233]: This filename looks too generic. What if we add searchIndex to other objects that would have different indexing fields? We can rename it to requestSearchIndex.json

File: pom.xml (hunk 1 above)
Comment (tatsiana-tarhonskaya) [reply pom.xml]: Updated request-storage to 5.1

File: ramls/request.json (hunk 2 above)
Comment (tatsiana-tarhonskaya) [reply ramls/request.json]: thanks, renamed it

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**